agent/src/beerocks/monitor/monitor_thread.cpp

```diff
@@ -1309,6 +1331,71 @@ bool monitor_thread::handle_cmdu(Socket *sd, ieee1905_1::CmduMessageRx &cmdu_rx)
     return true;
 }
 
+bool monitor_thread::handle_cmdu_ieee1905_1_message(Socket *sd, ieee1905_1::CmduMessageRx &cmdu_rx)
+{
+    auto cmdu_message_type = cmdu_rx.getMessageType();
+
+    switch (cmdu_message_type) {
+    case ieee1905_1::eMessageType::MULTI_AP_POLICY_CONFIG_REQUEST_MESSAGE:
+        return handle_multi_ap_policy_config_request(sd, cmdu_rx);
+    default:
+        LOG(ERROR) << "Unknown CMDU message type: " << std::hex << int(cmdu_message_type);
+    }
+
+    return false;
+}
+
+bool monitor_thread::handle_multi_ap_policy_config_request(Socket *sd,
+                                                           ieee1905_1::CmduMessageRx &cmdu_rx)
+{
+    /**
+     * The Multi-AP Policy Config Request message is sent by the controller, received and
+     * acknowledged by the backhaul manager, forwarded to the slave thread and forwarded back
+     * again to the monitor thread, where it is finally processed.
+     */
+    auto mid = cmdu_rx.getMessageId();
+    LOG(DEBUG) << "Received MULTI_AP_POLICY_CONFIG_REQUEST_MESSAGE, mid=" << std::hex << int(mid);
+
+    auto metric_reporting_policy_tlv = cmdu_rx.getClass<wfa_map::tlvMetricReportingPolicy>();
```

Comment:
`tlvMetricReportingPolicy` is optional?
If not, perhaps the function should fail right away:
```c++
if (!metric_reporting_policy_tlv) {
    LOG(ERROR) << "...";
    return false;
}

...
```

Reply:
Yes, it's optional